cpp/src/parquet/file_writer.cc

```diff
@@ -539,6 +540,7 @@ void WriteFileMetaData(const FileMetaData& file_metadata, ArrowOutputStream* sin
   metadata_len = static_cast<uint32_t>(position) - metadata_len;
 
   // Write Footer
+  metadata_len = ::arrow::bit_util::ToLittleEndian(metadata_len);
   PARQUET_THROW_NOT_OK(sink->Write(reinterpret_cast<uint8_t*>(&metadata_len), 4));
```

Comment:
```suggestion
{
  uint32_t metadata_len_le = ::arrow::bit_util::ToLittleEndian(metadata_len);
  PARQUET_THROW_NOT_OK(sink->Write(reinterpret_cast<uint8_t*>(&metadata_len_le), 4));
}
```

Can we solve like this?

Reply:
Sure.